### PR TITLE
archive(8.1) isolation: automated steps

### PR DESCRIPTION
**Pull request by pepopowitz**
## Description

Part of #3731.

Note: this PR will be merged into the `unsupported/8.1` branch. It will never make it into main.

Note: this PR will not go green. There are manual cleanup steps coming in a future PR that will make this branch green.

Continues the process of archival/isolation for version 8.1. Includes the following steps, which a computer did for me:

1. Exclude all other versions from version manifest files
2. Prevent the 8.1 site from being crawled
3. Fix the Dockerfile for local debugging
4. Update the swizzled theme components for the isolated version. 
5. Update CI workflows.
6. Update the docusaurus.config.js.
7. Configure the 3.9.0 optimize docs instance

See #2796 for a similar PR in a previous archived version, if that's helpful for comparison purposes during review.

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] There is **no urgency** with this change and can be released at any time.
- [x] My goal is to complete archival of 8.1 by the end of this week (May 3). I'd like to merge this in the next day to meet this schedule.

## PR Checklist

- [x] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [ ] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineerring DRI or delegate.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer.

### CODE REVIEW

**Review by akeller — APPROVED** [review]

**Comment by akeller** [comment]
I was second-guessing my approval because the build process was also modified in the linked reference PR, but I don't think that's necessary here. Can you confirm?

File: .github/workflows/publish-stage.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: publish-stage
 on:
   push:
     branches:
-      - "main"
```
Comment (pepopowitz) [inline .github/workflows/publish-stage.yaml:6]: This branch is going to be deployed to `stage.unsupported.docs.camunda.io/8.1`, and we only want to deploy this branch.

File: Dockerfile [hunk 2]
```diff
@@ -3,12 +3,12 @@
 FROM httpd:alpine
 
 RUN rm -r /usr/local/apache2/htdocs/*
-COPY /build /usr/local/apache2/htdocs/
+COPY /build /usr/local/apache2/htdocs/8.1/
```
Comment (pepopowitz) [inline Dockerfile:6]: This Dockerfile is for local development. Running the site within an 8.1 folder allows us to not have to make changes to the docusaurus configuration, which is expecting the site to be deployed within an 8.1 folder.

**Comment by pepopowitz** [comment]
As discussed over Slack, the issue is not the deployment workflows, but `build-docs.yaml` is missing a bunch of changes that were done [in the previous archival](https://github.com/camunda/camunda-docs/pull/2796/commits/09a426120e962991894ded6e368859fcdad46414).

This happened because the script to delete all that stuff is looking for the line to build the docs, which specifies the heap size.......which we updated recently, thus causing the script to not find what it was looking for.

Updating & adding that removal to this PR!

**Comment by pepopowitz** [comment]
> As discussed over Slack, the issue is not the deployment workflows, but `build-docs.yaml` is missing a bunch of changes that were done [in the previous archival](https://github.com/camunda/camunda-docs/pull/2796/commits/09a426120e962991894ded6e368859fcdad46414).
> 
> This happened because the script to delete all that stuff is looking for the line to build the docs, which specifies the heap size.......which we updated recently, thus causing the script to not find what it was looking for.
> 
> Updating & adding that removal to this PR!

`build-docs` workflow is fixed in latest commit. 

I'll update the associated script in a separate PR, so that it can target `main`.

**Comment by akeller** [comment]
Odd that it didn't prompt me to approve again, but you are good to merge when ready!

**Comment by pepopowitz** [comment]
> Odd that it didn't prompt me to approve again, but you are good to merge when ready!

Maybe GitHub doesn't care as much because it's not merging into `main`?

**Pull request merged**